### ISSUE TITLE
Error when the script calls functions

### ISSUE THREAD

**Issue by gamesfragers** [mattwilson1024/android-pattern-unlock#2]
Hi Matt,
I'm trying to get your script to run. But no matter where I call it from (adb on pc, android terminal) always gives me error: syntax error: 'WakeScreen

I have tried to see why but failed. I have a background in PHP so I understand logic and function syntax but I have no experience with shell script. I'm running your script on Nexus 2013 running Android 6 rooted and with ElementalX kernel.

Any help would be appreciated.

**Comment by mattwilson1024** [comment]
Hi @gamesfrager - the script works by executing a series of calls to adb, so you need to run it from your computer's terminal and not your phone's shell environment.

On Mac/Linux the script can be run from the terminal as per the usage comment at the top of the script. The Windows command prompt won't run shell scripts of this kind, instead try running it from git bash (this gets installed alongside Git for Windows). Alternatively you could try installing Cygwin and running it from there.

Also ensure that your phone has the "USB debugging" option turned on, and that you have adb in your PATH. Try running the "adb devices" command first - it should show you the ID of your connected device.

**Comment by gamesfragers** [comment]
First of all, thanks for the very quick reply.
I do have adb and connected properly to my device.

What I'm trying to do is unlock the pattern lock using sendevent using Tasker and run shell task.
I'm running this command: sh /sdcard/unlock.sh
I have just managed to do that actually!! and it worked after I realized that foe every x coordinate I had to multiply it by 1.12 and every y with 1.16 to scale it properly.

Now I'm facing the problem when screen is landscape.
I enabled show pointer location and I can see that I need to rotate my points, but I don't know how.

Anyway, thanks for the reply. I hope I can find a way to modify your script and make it work on android terminal.

Edit: what I mean by worked is that I managed to unlock usin just sendevent command not using your script. Like so:

# unlock.sh

#!/bin/sh
sendevent /dev/input/event0 3 57 14
sendevent /dev/input/event0 1 330 1
sendevent /dev/input/event0 3 53 403
sendevent /dev/input/event0 3 54 890
sendevent /dev/input/event0 3 58 57
sendevent /dev/input/event0 0 0 0
sendevent /dev/input/event0 3 53 907
sendevent /dev/input/event0 3 54 1009
sendevent /dev/input/event0 3 58 57
sendevent /dev/input/event0 0 0 0
......
it would be nice to have your script generate all these points automatically instead of manual input and run on my tablet's shell.

**Comment by mattwilson1024** [comment]
Happy to help. Sounds like an fun way of using it - I'll be interested to hear if you get it working. 

If you're running the script directly on the device, you probably need to remove "adb shell" from the front of each of the commands in my script, since in my case I was executing the script from my computer and trying to issue commands to the phone without having root access etc. From what you posted it's worth noting that the device string might need adjusting to suit your device as well.

E.g. try changing:
adb shell sendevent /dev/input/event2 3 53 $1
to:
sendevent /dev/input/event0 3 53 $1

and similar for the other lines in my script which run commands through adb shell.

**Comment by gamesfragers** [comment]
Yes, my touch screen is event0. I found that using getevent.
Then I got the scale factor by following http://stackoverflow.com/questions/16337656/get-android-sendevent-coordinates-scale-factor and used that in the script that I created manually.

I also had to comment each empty lines in your script. Apparently shell does not allow empty lines.
Then I had to make the for do loop into one line, also seems does not like returns!
I will keep experimenting. Thanks again.
